File: app/src/main/java/com/github/overpass/gather/screen/meeting/enrolled/EnrolledActivity.java
package com.github.overpass.gather.screen.meeting.enrolled;

import androidx.lifecycle.ViewModelProviders;

import com.airbnb.lottie.LottieAnimationView;
import com.github.overpass.gather.R;
import com.github.overpass.gather.screen.base.BaseActivity;

import butterknife.BindView;
import butterknife.OnClick;

public class EnrolledActivity extends BaseActivity<EnrolledViewModel> {

    @BindView(R.id.lavLargeTick)
    LottieAnimationView lavLargeTick;

    @Override
    protected int getLayoutRes() {
        return R.layout.activity_enrolled;
    }

    @Override
    protected EnrolledViewModel createViewModel() {
        return ViewModelProviders.of(this).get(EnrolledViewModel.class);
    }

    @Override
    protected void onBind() {
        if (viewModel.shouldAnimate()) {
            lavLargeTick.playAnimation();
        }
    }

    @OnClick(R.id.btnOk)
    public void onOkClick() {
        finish();
    }
}

File: app/src/main/java/com/github/overpass/gather/screen/meeting/enrolled/EnrolledActivity.kt
package com.github.overpass.gather.screen.meeting.enrolled

import android.os.Bundle
import androidx.lifecycle.Observer
import com.github.overpass.gather.App.Companion.appComponentManager
import com.github.overpass.gather.R
import com.github.overpass.gather.di.enrolled.EnrolledComponent
import com.github.overpass.gather.di.enrolled.EnrolledComponentManager
import com.github.overpass.gather.screen.base.BaseActivity
import kotlinx.android.synthetic.main.activity_enrolled.*

class EnrolledActivity : BaseActivity<EnrolledViewModel, EnrolledComponent>() {

    override val componentManager: EnrolledComponentManager
        get() = appComponentManager.getEnrolledComponentManager()

    override fun createComponent(): EnrolledComponent = componentManager.getOrCreate(Unit)

    override fun onComponentCreated(component: EnrolledComponent) {
        component.inject(this)
    }

    override val layoutRes: Int = R.layout.activity_enrolled

    override fun createViewModel(): EnrolledViewModel {
        return viewModelProvider.get(EnrolledViewModel::class.java)
    }

    override fun onBind() {
        viewModel.playAnimation().observe(this, Observer { nothing -> lavLargeTick.playAnimation() })
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        btnOk.setOnClickListener {
            finish()
        }
    }
}
